Firestore/core/test/firebase/firestore/testutil/async_testing.h

```diff
@@ -46,6 +52,84 @@ std::unique_ptr<util::Executor> ExecutorForTesting(const char* name);
  */
 std::shared_ptr<util::AsyncQueue> AsyncQueueForTesting();
 
+constexpr auto kTimeout = std::chrono::seconds(5);
+
+/**
+ * An expected outcome of an asynchronous test.
+ */
+class Expectation {
+ public:
+  Expectation();
+
+  /** Marks this expectation as fulfilled. */
```

Comment:
One thing that seems worthwhile to add to the comment is what happens if `Fulfill` is called more than once.